distributed/protocol/compression.py

```diff
@@ -93,6 +93,27 @@ def _fixed_lz4_decompress(data):
     }
     default_compression = "lz4"
 
+
+with ignoring(ImportError):
+    import zstandard as zstd
+
+    def zstd_compress(data):
+        if isinstance(data, (memoryview, bytearray)):
+            data = bytes(data)
+        return zstd.ZstdCompressor().compress(data)
```

Comment:
I think you don't need to create the compressor and decompressor on every compress call.

Also, the compressor can probably get more threads (like 2, which is also done for other compressors).

Furthermore, it would be nice if the compression level would be controllable via a config variable.

Reply:
+1

Reply:
New commit: #2970
https://github.com/dask/distributed/pull/2970/commits/8de39d744dcd96c2ddb94f0288e83a4358d1c191
 customizable compression level and many other options;

> I think you don't need to create the compressor and decompressor on every compress call.
Accepted.


> Also, the compressor can probably get more threads (like 2, which is also done for other compressors).
 threads=0, which means disabled multi-threaded compression. so it's safe in default cases.
    and also user can change the "distributed.comm.zstd.threads" value in dask.yaml  according to their requirements.


> Furthermore, it would be nice if the compression level would be controllable via a config variable.
comprehensive options are configurable now. includes compression level , import policy, ...